addons/crm/models/crm_lead.py

```diff
@@ -967,106 +989,60 @@ def _create_lead_partner_data(self, partner_name, is_company=False, parent_id=Fa
-            search_criteria = self.partner_name
-        elif self.contact_name:
-            search_criteria = self.contact_name
-        elif self.name:  # to be aligned with _create_lead_partner, search on lead's name as last possibility
-            search_criteria = self.name
-
-        if search_criteria:
-            return Partner.search([('name', 'ilike', '%' + search_criteria + '%')], limit=1).id
-
-        return False
-
-    def handle_partner_assignation(self, action='create', partner_id=False):
-        """ Handle partner assignation during a lead conversion.
+        if not partner:
+            # search through the existing partners based on the lead's partner or contact name
+            # to be aligned with _create_customer, search on lead's name as last possibility
+            for customer_potential_name in [self[field_name] for field_name in ['partner_name', 'contact_name', 'name'] if self[field_name]]:
+                partner = self.env['res.partner'].search([('name', 'ilike', '%' + customer_potential_name + '%')], limit=1)
+                if partner:
+                    break
 
-        If action is 'create', create new partner with contact and assign lead to
-        new partner_id. Otherwise assign lead to the specified partner_id
+        return partner
 
-        TDE FIXME: docstring does not match code... code seems a bit random.
+    def handle_partner_assignment(self, force_partner_id=False, create_missing=True):
+        """ Update customer (partner_id) of leads. Purpose is to set the same
+        partner on most leads; either through a newly created partner either
+        through a given partner_id.
 
-        :param string action: what has to be done regarding partners (create it, assign an existing one, or nothing)
-        :param int partner_id: partner to assign if any
-        :return: dict(lead_id: partner_id)
+        :param int force_partner_id: if set, update all leads to that customer;
+        :param create_missing: for leads without customer, create a new one
+          based on lead information;
         """
-        partner_ids = {}
         for lead in self:
-            if partner_id:
-                lead.partner_id = partner_id
-            if lead.partner_id:
-                partner_ids[lead.id] = lead.partner_id.id
-                continue
-            if action == 'create':
-                partner = lead._create_lead_partner()
-                partner_id = partner.id
-                partner.team_id = lead.team_id
-            partner_ids[lead.id] = partner_id
-        return partner_ids
-
-    def allocate_salesman(self, user_ids=None, team_id=False):
+            if force_partner_id:
+                lead.partner_id = force_partner_id
+            if not lead.partner_id and create_missing:
+                partner = lead._create_customer()
+                lead.partner_id = partner.id
+        return True
```

Comment:
its return True if everything went ok I guess. So it should return false if something went wrong ?
if not, why return something ? Especially, here, the returned value is never used (even in _convert_handle_partner, the return value is useless)

Reply:
Done